daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -2807,17 +2799,187 @@ def forward(node: ONNXOp, state: SDFGState,
+    C = out_desc_with_name(node, state, sdfg, "C")
+
+    # Create inner SDFG
+    op_sdfg = dace.SDFG(node.name)
+    compute_state = op_sdfg.add_state("compute_{}".format(op_name))
+
+    a_inner = A.clone()
+    a_inner.transient = False
+    b_inner = B.clone()
+    b_inner.transient = False
+    c_inner = C.clone()
+    c_inner.transient = False
+
+    op_sdfg.add_datadesc("A", a_inner)
+    op_sdfg.add_datadesc("B", b_inner)
+    op_sdfg.add_datadesc("C", c_inner)
+
+    # Create map and tasklet
+    map_ranges = {'__i%d' % i: '0:%s' % n for i, n in enumerate(A.shape)}
+    outer_me, outer_mx = compute_state.add_map(
+        '{}_map'.format(op_name),
+        map_ranges,
+        schedule=dace.ScheduleType.FPGA_Device)
+
+    tasklet = compute_state.add_tasklet(
+        '{}_task'.format(op_name), ['a_con', 'b_con'], ['c_con'],
+        'c_con = a_con {} b_con'.format(operator))
+
+    # Add data descriptors
+    a_in = compute_state.add_read("A")
+    b_in = compute_state.add_read("B")
+    c_out = compute_state.add_write("C")
+
+    # Connect map
+    if not streaming_node:
+        compute_state.add_memlet_path(
+            a_in,
+            outer_me,
+            tasklet,
+            dst_conn='a_con',
+            memlet=dace.Memlet("A[{}]".format(",".join(
+                ['__i%d' % i for i in range(len(A.shape))]))))
+
+        compute_state.add_memlet_path(
+            b_in,
+            outer_me,
+            tasklet,
+            dst_conn='b_con',
+            memlet=dace.Memlet("B[{}]".format(",".join(
+                ['__i%d' % i for i in range(len(B.shape))]))))
+
+        compute_state.add_memlet_path(
+            tasklet,
+            outer_mx,
+            c_out,
+            src_conn='c_con',
+            memlet=dace.Memlet("C[{}]".format(",".join(
+                ['__i%d' % i for i in range(len(C.shape))]))))
+
+    else:
+        #memlet from stream
+        compute_state.add_memlet_path(a_in,
+                                      outer_me,
+                                      tasklet,
+                                      dst_conn='a_con',
+                                      memlet=dace.Memlet("A[0,0,0,0]"))
+
+        #memlet from stream
+        compute_state.add_memlet_path(b_in,
+                                      outer_me,
+                                      tasklet,
+                                      dst_conn='b_con',
+                                      memlet=dace.Memlet("B[0,0,0,0]"))
+
+        compute_state.add_memlet_path(tasklet,
+                                      outer_mx,
+                                      c_out,
+                                      src_conn='c_con',
+                                      memlet=dace.Memlet("C[0,0,0,0]"))
+
+    return op_sdfg
+
+
+def binary_operator_general_checks(node: ONNXOp, state: SDFGState,
+                                   sdfg: SDFG) -> bool:
+
+    A = in_desc_with_name(node, state, sdfg, "A")
```

Comment:
Broadcasting binary operators is ubiquitous in DL applications. Please make sure you either support this, or return false in the general checks